py/obj.c

```diff
@@ -235,12 +235,17 @@ mp_int_t mp_obj_get_int(mp_const_obj_t arg) {
     } else if (MP_OBJ_IS_TYPE(arg, &mp_type_int)) {
         return mp_obj_int_get_checked(arg);
     } else {
+        #if MICROPY_PY_ALL_SPECIAL_METHODS
+        mp_obj_t res = mp_unary_op(MP_UNARY_OP_INT, (mp_obj_t)arg);
+        return mp_obj_int_get_checked(res);
```

Comment:
If the return value of `mp_unary_op` is not an integer (small or big) then this will segfault.

Reply:
> If the return value of mp_unary_op is not an integer (small or big) then this will segfault.

So, do we have similar checks everywhere by now? I can add that check to mp_unary_op(), too bad we don't have a define to disable such checks. (MICROPY_FULL_CHECKS is described to be crash-safe.)